frame/staking/src/pallet/mod.rs

```diff
@@ -439,6 +476,61 @@ pub mod pallet {
 		ValueQuery,
 	>;
 
+	/// The nominator count each `ExposurePage` is capped at.
+	///
+	/// This cannot be greater than `T::MaxNominatorRewardedPerValidator`.
+	#[pallet::storage]
+	pub type ExposurePageSize<T> = StorageValue<_, u32, OptionQuery>;
+
+	/// Maximum number of exposure pages that can be stored for a single validator in an era.
+	///
+	/// Must be greater than 0.
+	///
+	/// When this is set to 1, the reward payout behaviour is similar to how it used to work before
+	/// we had paged exposures.
+	#[pallet::storage]
+	// TODO(ank4n): Does it need #[pallet::getter(...)]
+	pub type MaxExposurePageCount<T> = StorageValue<_, PageIndex, OptionQuery>;
+
+	/// Paginated exposure of a validator at given era.
+	///
+	/// This is keyed first by the era index to allow bulk deletion, then the tuple of stash account
+	/// and page.
+	///
+	/// This uses DoubleMap instead of NMap to efficiently clear this after `HISTORY_DEPTH` eras.
+	/// If stakers hasn't been set or has been removed then empty exposure is returned.
+	#[pallet::storage]
+	#[pallet::getter(fn eras_stakers_paged)]
+	#[pallet::unbounded]
+	pub type ErasStakersPaged<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		EraIndex,
+		Twox64Concat,
+		(T::AccountId, PageIndex),
```

Comment:
I think there is a good advantage if we use an N-MAP here: We can easily find the number of pages (possible third keys) by knowing the first two key parts. I suppose that's useful.